assets/app/components/siteContainer.js

```diff
@@ -21,7 +21,23 @@ class SiteContainer extends React.Component {
   }
 
   componentDidMount() {
-    const { storeState, params, routeParams } = this.props;
+    const { storeState } = this.props;
+    if (storeState.sites.length) {
+      this.downloadCurrentSiteData();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { storeState} = this.props;
```

Comment:
This is my only comment. everything else looks good since we don't have any coverage on this pesky component yet.